File: articles/robust-decision-making/r/run_all_robust_decision_workflows.R
args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)
article_root <- if (length(file_arg) > 0) normalizePath(file.path(dirname(normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)), ".."), mustWork = TRUE) else getwd()
r_dir <- file.path(article_root, "r")
scripts <- c("robust_decision_making_workflow.R", "robustness_tables.R", "regret_tables.R", "vulnerability_tables.R", "threshold_review_tables.R", "robust_decision_review_summary.R")
for (script in scripts) { path <- file.path(r_dir, script); message("Running ", path); source(path) }
message("All R robust decision workflows completed.")

File: articles/robust-decision-making/python/run_all_robust_decision_workflows.py
#!/usr/bin/env python3
from __future__ import annotations
from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"
SCRIPTS = [
    "expected_value_model.py",
    "worst_case_analysis.py",
    "regret_analysis.py",
    "threshold_compliance.py",
    "vulnerability_analysis.py",
    "adaptive_trigger_review.py",
    "decision_record_exporter.py",
    "robust_decision_making_simulation.py",
]

def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python robust decision workflows completed.")

if __name__ == "__main__":
    main()
